### DISCUSSION TITLE
Is It Possible To Run Go Programs Inside Godot?

### DISCUSSION THREAD

**Discussion by AvicennaJr** [godot-go/godot-go#72] [Q&A]
I was thinking of creating a windows application for a go program I built. The go program is a programming language which is an extension of the book "Writing an Interpreter in Go" by Thorsten Bell.

The general idea is to create an IDE for the custom language, it would be a simpler version of the python IDE for windows. So is making something like that possible with the help of this package?

**Comment by pcting** [comment]
it is possible, but i wouldn't build anything on top of this project until this is resolved: https://github.com/godot-go/godot-go/issues/71

but it should be possible for what you're asking for as you have a full go environment loaded in as a shared library in the godot process.

**Reply by AvicennaJr** [reply]
Thanks, will wait for the issue to be resolved and I'll try

**Comment by AvicennaJr** [comment]
I put my original project on halt, I'll probably use something like wails for my use case.

**Comment by pcting** [comment]
just wanted to update everyone that godot-go is now running against godot 4.2-dev6. the [Dodge the Creeps](https://github.com/godot-go/godot-go-demo-projects/tree/main/2d/dodge_the_creeps) demo has been updated and runs without crashes on my linux machine.

going to close this out for now.

cheers
